assignments/closure.js

```diff
@@ -1,21 +1,54 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+function multiFunc(x) {
+  return function(y) {
+    return x * y;
+  }
+}
 
+const by5 = multiFunc(5);
+const by10 = multiFunc(10);
+
+console.log(by10(2));
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Create a counter function ====
+
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
+  let count = 0;
+  return function() {
+    return count += 1;
+  }
 };
+
+const newCounter = counter();
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
 // ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
+let counter = 0;
 const counterFactory = () => {
   // Return an object that has two methods called `increment` and `decrement`.
   // `increment` should increment a counter variable in closure scope and return it.
   // `decrement` should decrement the counter variable and return it.
-};
+  return {
+    'increment' : function() {
+      return counter += 1;
+    },
+    'decrement' : function() {
+      return counter -= 1;
+    }
+  }
+
+}
+
+const masYMenos = counterFactory();
+console.log(masYMenos.increment());
+console.log(masYMenos.increment());
+console.log(masYmMnos.decrement());
+
```

Comment:
MDN has a really good practical example of how to use closures while keeping the counter variable hidden from global scope.  Here's the code : 
![image](https://user-images.githubusercontent.com/30554629/59475039-04576200-8e0f-11e9-8ba8-a4883f99a36c.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures